components/dropdown/script.js

```diff
@@ -821,7 +823,15 @@ class GamefaceDropdown extends CustomElementValidator {
         if (!event.ctrlKey) this.selected = null;
         if (option.isAlreadySelected(this, option)) return this.deselect(option);
 
-        this.setSelectedAndScroll(option);
+        if (event.shiftKey) {
+            const fromIdx = this.enabledOptions.indexOf(this._pivotIndex);
+            const toIdx = Array.from(this.allOptions).indexOf(option);
+            const direction = (fromIdx - toIdx) > 0 ? -1 : 1;
+
+            this.selectFromTo(fromIdx, this.enabledOptions.indexOf(toIdx), direction, true, false);
+        }
```

Comment:
To prevent multiple calls of document.queryselector that is done in the getters.

```suggestion
        if (event.shiftKey) {
            const enabledOptions = this.enabledOptions;
            const fromIdx = enabledOptions.indexOf(this._pivotIndex);
            const toIdx = Array.from(this.allOptions).indexOf(option);
            const direction = (fromIdx - toIdx) > 0 ? -1 : 1;

            this.selectFromTo(fromIdx, enabledOptions.indexOf(toIdx), direction, true, false);
        }
```